### PR TITLE
Install iothub_client internal headers ALSO under azureiot/internal

### PR DESCRIPTION
**Pull request by ewertons**
# Checklist
- [x] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-c/blob/main/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [ ] I edited the corresponding document in the `devdoc` folder and added or modified requirements.
- I submitted this PR against the correct branch: 
  - [x] This pull-request is submitted against the `main` branch. 
  - [x] I have merged the latest `main` branch prior to submission and re-merged as needed after I took any feedback.
  - [x] I have squashed my changes into one with a clear description of the change.

# Reference/Link to the issue solved with this PR (if any)

# Description of the problem

ARM and MIPS32 crosscompilation builds are failing due to changes in upload-to-blob header includes throughout azure-iot-sdk-c.

# Description of the solution
 
- Install iothub_client internal headers ALSO under "azureiot/internal", since many #include use the "internal" relative path.
- Add "include/azureiot" to include_directories of ARM and MIPS32 toolchains, since that's where the headers is installed by default.

Note: These tasks (cross-comp of ARM and MIPS32) apparently ran when the PR for upload to blob changes was validated for merging, but they test against main in https://github.com/azure/azure-iot-sdk-c, not against the PR code itself.

### CODE REVIEW

File: iothub_client/CMakeLists.txt [hunk 1]
```diff
@@ -76,23 +80,28 @@ if (use_edge_modules)
...
+    set(iothub_client_internal_h_files
+        ${iothub_client_internal_h_files}
         ${CMAKE_CURRENT_LIST_DIR}/inc/internal/iothub_client_hsm_ll.h
     )
 endif()
 #end back compat only
 
 set(iothub_client_h_install_files
     ${iothub_client_h_files}
+    ${iothub_client_internal_h_files}
```
Comment (vaavva) [inline iothub_client/CMakeLists.txt:100]: Isn't `iothub_client_internal_h_files` included in `iothub_client_h_files`?

**Review by vaavva — APPROVED** [review]

File: samples/dockerbuilds/ARM/Dockerfile [hunk 2]
```diff
@@ -91,6 +94,7 @@ RUN echo "SET(CURL_LIBRARIES ${TOOLCHAIN_PREFIX}/lib/libcurl.so)" >> toolchain.c
 RUN echo "SET(ENV{LDFLAGS} -L${TOOLCHAIN_PREFIX}/lib)" >> toolchain.cmake
 RUN echo "SET(set_trusted_cert_in_samples true CACHE BOOL \"Force use of TrustedCerts option\" FORCE)" >> toolchain.cmake
 RUN echo "include_directories(${TOOLCHAIN_PREFIX}/include)" >> toolchain.cmake
+RUN echo "include_directories(${TOOLCHAIN_PREFIX}/include/azureiot)" >> toolchain.cmake
```
Comment (CIPop) [inline samples/dockerbuilds/ARM/Dockerfile:97]: I don't understand: how was this working in the past? If we change the install directory, wouldn't all samples require something like `#include <azureiot/...>` before and now without azureiot?

**Review by CIPop — COMMENTED** [review]
If the files were installing in `include/` before and will now install in `azureiot/include`, that is a breaking API change as existing applications will either need to change their INCLUDE definitions (as this PR does), or their sources, to prepend `azureiot/` to each `#include ` statement.

File: iothub_client/CMakeLists.txt (hunk 1 above)
Comment (ewertons) [reply iothub_client/CMakeLists.txt:100]: yes, not to break the existing customers.

File: samples/dockerbuilds/ARM/Dockerfile (hunk 2 above)
Comment (ewertons) [reply samples/dockerbuilds/ARM/Dockerfile:97]: That's something I was not able to pinpoint.

**Comment by ewertons** [comment]
This PR will be reworked.

**Pull request closed without merging**